catalog/ui/catalog-ui-search/src/main/webapp/component/query-annotations/query-annotations.view.js

```diff
@@ -0,0 +1,225 @@
+            notesList: '.notes-list'
+        },
+        _notesCollection: undefined,
+        _notes: undefined,
+        initialize: function (options) {
+            this.selectionInterface = options.selectionInterface || this.selectionInterface;
+            if (!options.model) {
+                this.setDefaultModel();
+            }
+        },
+        onBeforeShow: function(){
+            this.clearNotes();
+            this.showAddNoteView();
+            this.getNotesForMetacard();
+            this.showNotesListView();
+            this.turnOnEditing();
+            this.setupListeners();
+        },
+        clearNotes: function () {
+            if (!this._notesCollection) {
+                this._notesCollection = new NoteCollection();
+            }
+            this._notesCollection.reset();
+        },
+        showAddNoteView: function() {
+            this.addNoteField.show(new PropertyView({
+                model: new Property({
+                    value:[''],
+                    placeholder:'enter note',
+                    type:'TEXTAREA',
+                    label: ''
+                })
+            }));
+        },
+        getNotesForMetacard: function () {
+            LoadingCompanionView.beginLoading(this);
+                $.get('/search/catalog/internal/notes/' + this.model.get('id') + "?isVirtual=true")
+                    .then(function (response) {
+                        var resp = response.response;
+                        if (response.responseType === "success") {
+                            if (this.isValidResponse(resp)) {
+                                this._notes = JSON.parse(resp);
+                                this.parseNotes();
+                            } else {
+                                announcement.announce({
+                                    title: 'Error!',
+                                    message: "There was an error retrieving the notes for this item!",
+                                    type: 'error'
+                                });
+                            }
+                        }
+                        LoadingCompanionView.endLoading(this);
+                        this.checkHasNotes();
+                    }.bind(this));
+        },
+        showNotesListView: function() {
+            this.notesList.show(new NoteCollectionView({
+                collection: this._notesCollection,
+                selectionInterface: this.selectionInterface,
+                parent: this.model
+            }));
+            this.notesList.currentView.$el.addClass("is-list");
+        },
+        setupListeners: function() {
+            this.listenTo(this._notesCollection, "remove", function() {
+                this.checkHasNotes();
+            });
+        },
+        handleRefresh: function() {
+            this.getNotesForMetacard();
+            announcement.announce({
+                            title: 'Success!',
```

Comment:
My current version looks OK, but I'm not 100% about the JS formatting rules. I re-ran the code formatter, and it didn't change anything.